android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/AuthenticationManager.java

```diff
@@ -533,13 +533,15 @@ public Completable requestPhoneSignIn(@NonNull String regionCode, @NonNull Strin
                             ConsentStatus consentStatus = consentStatusEntry.getValue();
 
                             // required consent missing on Bridge and present locally
-                            if (consentStatus.isRequired()
-                                    && !consentStatus.isConsented()
-                                    && isConsentedInLocal(subpopulationGuid)) {
-
-                                // upload local consents, fail this sign in if consent upload fails
-                                return uploadLocalConsents()
-                                        .toSingle();
+                            if (consentStatus.isRequired() && !consentStatus.isConsented()) {
+                                if (isConsentedInLocal(subpopulationGuid)) {
+                                    // upload local consents, fail this sign in if consent upload fails
+                                    return uploadLocalConsents().toSingle();
+                                } else {
+                                    // No saved consents for the required guid, propagate the error
+                                    return Single.error(new ConsentRequiredException
```

Comment:
Alternately, this code could be moved to  onErrorResumeNext and you can just propagate the original exception